lib/eex/lib/eex.ex

```diff
@@ -166,15 +166,18 @@ defmodule EEx do
   Gets a string `source` and generate a quoted expression
   that can be evaluated by Elixir or compiled to a function.
   """
-  def compile_string(source, options \\ []) do
+  @spec compile_string(String.t, Keyword.t) :: Macro.t | no_return
+  def compile_string(source, options \\ []) when is_binary(source) and is_list(options) do
     EEx.Compiler.compile(source, options)
   end
 
   @doc """
   Gets a `filename` and generate a quoted expression
   that can be evaluated by Elixir or compiled to a function.
   """
-  def compile_file(filename, options \\ []) do
+  # TODO: REVIEW if `filename` should be `String.t`
```

Comment:
Actually, no. It should be `String.t` because below we assume the filename is a string. So no `Path.t` in EEx. We should add a guard too.